### ISSUE TITLE
test_birkhoff_von_neumann_decomposition failure

### ISSUE THREAD

**Issue by tdb4197** [jfinkels/birkhoff#3]
Running on Fedora 27

pip3 install nose networkx birkhoff numpy --user

Run the test_birkhoff_von_neumann_decomposition() function with the following result

```python
>>> test.test_birkhoff_von_neumann_decomposition()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/tbottom/test.py", line 55, in test_birkhoff_von_neumann_decomposition
    actual = birkhoff_von_neumann_decomposition(D)
  File "/home/tbottom/.local/lib/python3.6/site-packages/birkhoff/decompose.py", line 173, in birkhoff_von_neumann_decomposition
    M = hopcroft_karp_matching(G)
  File "/home/tbottom/.local/lib/python3.6/site-packages/birkhoff/matching.py", line 98, in hopcroft_karp_matching
    left, right = bipartite_sets(G)
  File "/home/tbottom/.local/lib/python3.6/site-packages/networkx/algorithms/bipartite/basic.py", line 203, in sets
    raise nx.AmbiguousSolution(msg)
networkx.exception.AmbiguousSolution: Disconnected graph: Ambiguous solution for bipartite sets.
```

**Comment by jfinkels** [comment]
This is probably a bug due to a change in NetworkX, the package that handles the [bipartite graph matching algorithm][1]. This would probably be resolved by providing the "left" set of nodes as a second argument to the `networkx.algorithms.bipartite.matching.maximum_matching` function. In this case, we know the left and right node sets because we are constructing them: the left set of nodes is {0, …, *n* - 1} and the right set is {*n*, …, 2*n* - 1} (or vice versa).

[1]: https://networkx.github.io/documentation/stable/reference/algorithms/generated/networkx.algorithms.bipartite.matching.hopcroft_karp_matching.html#networkx.algorithms.bipartite.matching.hopcroft_karp_matching

**Comment by jfinkels** [comment]
Fixed by pull request #4.
